### PR TITLE
Make avatar colors depend on user ID

### PR DESCRIPTION
**Pull request by pferreir**
Also added 6 new colors to increase variety.

image

### CODE REVIEW

File: indico/modules/users/util.py [hunk 1]
```diff
@@ -379,8 +379,15 @@ def merge_users(source, target, force=False):
     logger.info('Successfully merged %s into %s', source, target)
 
 
-def get_color_for_username(username):
-    return user_colors[crc32(username) % len(user_colors)]
+def get_color_for_user_id(user_id):
+    """Calculate a unique color for a user based on their id.
+
+        :param user_id: the user ID (int), or otherwise a string (external search results)
```
Comment (ThiefMaster) [inline indico/modules/users/util.py:385]: Just wondering.. wouldn't it be simpler to just use `crc32(str(user_id))` whichw ould work for all kinds of ids?

File: indico/modules/users/util.py [hunk 2]
```diff
@@ -421,10 +428,17 @@ def set_user_avatar(user, avatar, filename, lastmod=None):
...
-        color = get_color_for_username(name)
+def send_default_avatar(user):
+    """Send a user's default avatar as an SVG.
+
+       :param user: A `User` object or string (external search results, registrations)
+    """
+    if isinstance(user, str):
+        text = user[0].upper()
+        color = get_color_for_user_id(user)
+    elif user.full_name:
```
Comment (ThiefMaster) [inline indico/modules/users/util.py:439]: That will fail when called via `/assets/avatar/blank.svg` since user is None

File: indico/modules/users/util.py [hunk 3]
```diff
@@ -421,10 +428,17 @@ def set_user_avatar(user, avatar, filename, lastmod=None):
...
 
 
-def send_default_avatar(name=None):
-    if name:
-        text = name[0].upper()
-        color = get_color_for_username(name)
+def send_default_avatar(user):
+    """Send a user's default avatar as an SVG.
+
+       :param user: A `User` object or string (external search results, registrations)
```
Comment (ThiefMaster) [inline indico/modules/users/util.py:434]:
```suggestion
    :param user: A `User` object or string (external search results, registrations)
```

File: indico/modules/users/util.py (hunk 2 above)
Comment (pferreir) [reply indico/modules/users/util.py:439]: Oh. I didn't test that edge case. When does it happen?

File: indico/modules/users/util.py (hunk 1 above)
Comment (pferreir) [reply indico/modules/users/util.py]: Yeah, I thought of it, but kind of felt dirty doing it since `user_id` could be anything and result in unexpected behavior. But it's not a bad solution either.

File: indico/modules/users/util.py (hunk 2 above)
Comment (ThiefMaster) [reply indico/modules/users/util.py:439]: anonymous users in editing timelines for example (via `_get_anonymous_user` when it's configured to not reveal the name of editors to submitters)

possibly for a pending user with no first name as well (there's a moment where you have a pending user without a user id: f90b120a135)

File: indico/modules/users/util.py (hunk 1 above)
Comment (pferreir) [reply indico/modules/users/util.py]: Sometimes I wish Python was a strongly-typed language 🙊

Comment (ThiefMaster) [reply indico/modules/users/util.py]: you could use `def get_color_for_user_id(user_id: t.Union[int, str])` ;)
still not really strongly-typed of course, but sphinx uses this information when autodocumenting a function for example

**Review by ThiefMaster — APPROVED** [review]
